torch/ao/quantization/fx/_model_report/model_report_visualizer.py

```diff
@@ -102,135 +112,256 @@ def get_all_unique_feature_names(self, plottable_features_only: bool = True) ->
+            statistics are global, and which are actually per-channel, so it's better to split it up into two
+            tables. This also makes the information much easier to digest given the plethora of statistics collected
 
-        Table columns:
+        Tensor table columns:
+            idx  layer_fqn  feature_1   feature_2   feature_3   .... feature_n
+            ----  ---------  ---------   ---------   ---------        ---------
 
-         idx  layer_fqn   type  shape  feature_1   feature_2   feature_3   .... feature_n
-        ----  ---------   ----  -----  ---------   ---------   ---------        ---------
+        Per-Channel table columns:
+            idx  layer_fqn  channel  feature_1   feature_2   feature_3   .... feature_n
+            ----  ---------  -------  ---------   ---------   ---------        ---------
 
         Args:
-            feature (str, optional): The specific feature we wish to generate the table for
+            feature_filter (str, optional): Filters the features presented to only those that
+                contain this filter substring
                 Default = "", results in all the features being printed
-            module_fqn_prefix_filter (str, optional): Only includes modules with this string prefix
+            module_fqn_filter (str, optional): Only includes modules with this string
                 Default = "", results in all the modules in the reports to be visible in the table
 
-        Expected Use:
-            >>> model_report_visualizer.generate_table_visualization(*filters)  # outputs neatly formatted table
+        Returns a dictionary with two keys:
+            (Dict[strList[List[Any]]]) A dict containing two keys:
+            "tensor_level_info", "channel_level_info"
+                Each key maps to:
+                    A list of lists containing the table information row by row
+                    The 0th index row will contain the headers of the columns
+                    The rest of the rows will contain data
         """
-        pass
-
-    def generate_plot_info(self, feature: str, module_fqn_prefix_filter: str = "") -> Tuple[Callable, List[List[Any]]]:
+        # first get the filtered data
+        filtered_data: OrderedDict[str, Any] = self._get_filtered_data(feature_filter, module_fqn_filter)
+
+        # now we split into tensor and per-channel data
+        tensor_features: Set[str] = set()
+        channel_features: Set[str] = set()
+
+        # keep track of the number of channels we have
+        num_channels: int = 0
+
+        for module_fqn in filtered_data:
+            for feature_name in filtered_data[module_fqn]:
+                # get the data for that specific feature
+                feature_data = filtered_data[module_fqn][feature_name]
+
+                # if it is only a single value, is tensor, otherwise per channel
+                try:
+                    iterator = iter(feature_data)
+                except TypeError:
```

Comment:
is there an easier way to check? e.g. through types? `using try/catch to check something is not recommended`

Reply:
Yeah I looked up a bunch of things on stackoverflow and other sites and the recommended method everybody said to use for Python specifically is try catch, which people are calling the "pythonic" way of doing things even though it's not really popular with other languages.

After looking up multiple resources, it doesn't look like there is a more elegant way for now. I tried making it checking for types specifically but i got errors like 

```
    raise TypeError("len() of a 0-d tensor")
TypeError: len() of a 0-d tensor
```
in my tests because you can have a single value tensor.

Reply:
how are you checking for types?
does `isinstance(feature_data, list)` work?

Reply:
this is what i tested but because some of the values are lists, and others are tensors, this alone doesn't work. And I also tried `isinstance(feature_data, torch.Tensor)` but that didn't work either because of single value tensors. Updated to have if else statements instead.